reana_client/cli/workflow.py

```diff
@@ -787,18 +786,25 @@ def add_verbose_data_from_response(response, verbose_headers, headers, data):
             headers += ["duration"]
             data[-1] += [get_workflow_duration(workflow_response) or "-"]
 
-        if output_format:
-            tablib_data = tablib.Dataset()
-            tablib_data.headers = headers
-            for row in data:
-                tablib_data.append(row)
-
-            if _format:
-                tablib_data = tablib_data.subset(rows=None, cols=list(_format))
+        tablib_data = tablib.Dataset()
+        tablib_data.headers = headers
+        for row in data:
+            tablib_data.append(row=row, tags=row)
 
-            display_message(tablib_data.export(output_format))
+        if _format:
+            tablib_data, filtered_headers = format_data(
+                parsed_format_filters, headers, tablib_data
+            )
+            if output_format:
+                display_message(json.dumps(tablib_data))
+            else:
+                tablib_data = [list(item.values()) for item in tablib_data]
+                click_table_printer(filtered_headers, filtered_headers, tablib_data)
         else:
-            click_table_printer(headers, _format, data)
+            if output_format:
+                display_message(tablib_data.export(output_format))
+            else:
+                click_table_printer(headers, _format, data)
```

Comment:
There are (small variations of) this code in a few commands of reana-client. Should we open an issue to see if this can be improved?

Reply:
Yes, created it here: https://github.com/reanahub/reana-client/issues/640